crates/runtime/src/search/rrf.rs

```diff
@@ -178,6 +305,50 @@ impl ReciprocalRankFusion {
             return exec_err!("{RRF_UDF_NAME} unable to compute fused_score");
         };
 
+        // If user specifies a recency column, we enable recency boosting
+        let qualified_recency_col = if let Some(recency_col) = args.time_column.clone() {
+            let (_, qname) = recency_col.qualified_name();
+            Self::first_qualified_field(&subquery_dfs[0], &qname)?
+        } else {
+            return Ok(score_expr);
+        };
+
+        // Defaults: exponential decay over days (86400s)
+        let recency_decay = args
+            .recency_decay
+            .clone()
+            .unwrap_or(RecencyDecay::Exponential);
+        let decay_scale_secs = args.decay_scale_secs.unwrap_or(86400.0);
+
+        // Lots of casting annoyances are avoided by treating everything as `long`
+        let today_epoch = to_unixtime(vec![now()]);
+        let recency_col_epoch = to_unixtime(vec![col(qualified_recency_col)]);
+        let age_in_units = (today_epoch - recency_col_epoch) / lit(decay_scale_secs);
+
+        let recency_expr = match recency_decay {
+            // e^(-alpha * age units)
+            RecencyDecay::Exponential => {
+                let decay_constant = args.decay_constant.unwrap_or(0.01);
+                #[allow(clippy::neg_multiply)]
```

Comment:
I know. Why is this guy always disabling the linting rules. 

For this one: I am too inattentive to notice a little dash, and so the explicit multiply by `-1` makes it clearer to me that it's there vs `-decay_constant`.